### PR TITLE
Initial snap packaging.

### LINKED ISSUE #152: [Request] Make Software app available as snap
For an alpha application, it works pretty well. If possible please make it available as a snap.
This way it will be easier to keep up with new releases and bug fixes.

### PR DESCRIPTION
**Pull request by oSoMoN**
Fixes #152.

### CODE REVIEW

**Comment by Feichtmeier** [comment]
Awesome 😎👍
I'm not competent enough to review this. So is it ready to merge? :)

**Comment by kenvandine** [comment]
> Awesome sunglasses+1 I'm not competent enough to review this. So is it ready to merge? :)

I'll review this

**Comment by oSoMoN** [comment]
> So is it ready to merge? :)

It works well enough in my shallow local tests, so I'd say yes, it's ready to merge, provided it passes Ken's review.

File: snap/snapcraft.yaml [hunk 1]
```diff
@@ -0,0 +1,82 @@
+name: software
```
Comment (kenvandine) [inline snap/snapcraft.yaml:1]: This will be published as snap-store, so the snap name should be "snap-store"

File: snap/snapcraft.yaml [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+adopt-info: software
+summary: Ubuntu Software
+description: An Ubuntu software app made with Flutter.
```
Comment (kenvandine) [inline snap/snapcraft.yaml:4]: This should be made a bit richer, and the fact that it's a flutter app isn't actually useful.  We can start with the description in the snap-store, perhaps tweaking it to make sure it accurately reflects features currently available.

File: snap/snapcraft.yaml [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+apps:
+  software:
+    command: bin/software
+    desktop: bin/data/flutter_assets/assets/software.desktop
+    extensions: [gnome]
+    plugs:
+      - appstream-metadata
+      - desktop
+      - desktop-legacy
+      - dot-snap-auth-json
```
Comment (kenvandine) [inline snap/snapcraft.yaml:60]: I don't think we need dot-snap-auth-json, at least not at this point.

**Review by kenvandine — CHANGES_REQUESTED** [review]
Added a few comments

**Review by kenvandine — APPROVED** [review]
Looks good, thanks!

**Pull request merged**